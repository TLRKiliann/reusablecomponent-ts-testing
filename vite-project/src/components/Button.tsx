import { Component } from 'react'

interface DataProps {
  isClicked: boolean
  description: string
}

export default class Button extends Component<DataProps> {
  state = {
      isClicked: false,
      description: "it's false"
  }

  handleClick = (event: MouseEvent<HTMButtonElement>) => {
    {this.state.isClicked 
      ? this.setState({isClicked: false, description: "it's false"})
      : this.setState({isClicked: true, description: "it's true"})}
  }

  render() {
    return(
      <>
      <button data-testid="btnid" onClick={this.handleClick} >
        my btn
      </button>
      <input
        data-testid="inputid"
        value={this.state.description} 
        onChange={this.handleClick}
        placeholder="inputplace"
      />
    </>
    )
  }
}
