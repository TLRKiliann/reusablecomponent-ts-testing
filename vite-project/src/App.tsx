import { useState, Component } from 'react'
import Header from './components/Header'
import Button from './components/Button'
import TogetherComp from './components/TogetherComp'
import './App.css'

class App extends Component<{}> {
  render() {
    return (
      <div className="App">

        <Header />

        <Button />

        <TogetherComp />

      </div>
    )
  }
}

export default App
