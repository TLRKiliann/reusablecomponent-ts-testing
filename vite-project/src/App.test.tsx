import {beforeEach, describe, expect, test, it, vi} from 'vitest';
import {fireEvent, render, screen, waitFor} from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import {create} from 'react-test-renderer';
import App from './App.tsx'
import Header from './App.tsx'
import Button from './App.tsx'
import Together from './App.tsx'


describe('My first snapshot test', ()=> {
    test('testing app button', () => {
      let tree = create(<App />)
      expect(tree.toJSON()).toMatchSnapshot();
    })
})

describe('Third snapshot test', () => {
	test('testing Header Comp test', () => {
		let tree = create(<Header />)
		expect(tree.toJSON()).toMatchSnapshot();
	})
})

describe('Second snapshot test', ()=> {
    test('testing BTN component reusable', () => {
      let tree = create(<Button />)
      expect(tree.toJSON()).toMatchSnapshot();
    })
})

describe('Fourth snapshot test', () => {
	test('testing Together component', () => {
		let tree = create(<Together />)
		expect(tree.toJSON()).toMatchSnapshot();
	})
})

describe('Button', () => {
  const description = "it's true";
  beforeEach(() => {
    console.log("beforeEach");
  })

  afterEach(() => {
    console.log("afterEach");
  })

  afterAll(() => {
    console.log("After All Tests");
  })
  test('should render a description object', async () => {
    const { debug, getByTestId } = render(<Button />)
    const itemName = await screen.getAllByText<string>(/my btn/i);
    debug()
  })
});