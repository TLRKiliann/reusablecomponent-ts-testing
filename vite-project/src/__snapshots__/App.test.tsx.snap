// Vitest Snapshot v1

exports[`Fourth snapshot test > testing Together component 1`] = `
<div
  className="App"
>
  <div>
    Hello from Header
  </div>
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />
  <div>
    Hello from Header
  </div>
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />
</div>
`;

exports[`My first snapshot test > testing app button 1`] = `
<div
  className="App"
>
  <div>
    Hello from Header
  </div>
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />
  <div>
    Hello from Header
  </div>
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />
</div>
`;

exports[`Second snapshot test > testing BTN component reusable 1`] = `
<div
  className="App"
>
  <div>
    Hello from Header
  </div>
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />
  <div>
    Hello from Header
  </div>
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />
</div>
`;

exports[`Third snapshot test > testing Header Comp test 1`] = `
<div
  className="App"
>
  <div>
    Hello from Header
  </div>
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />
  <div>
    Hello from Header
  </div>
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />
</div>
`;
